c_api/tests/c_api_behavior_test.cpp

```diff
@@ -113,3 +113,504 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
 
   inner(TI_ARCH_VULKAN);
 }
+
+TEST_F(CapiTest, TestBehaviorAllocateMemory) {
+  TiError error = TI_ERROR_SUCCESS;
+
+  auto inner = [&](TiArch arch) {
+    if (ti::is_arch_available(arch)) {
```

Comment:
Is there a particular reason we have to stuff everything in this branch? See the impl of `TestBehaviorGetRuntimeCapabilities`.